airflow/operators/python_operator.py

```diff
@@ -86,6 +87,38 @@ def __init__(
         if templates_exts:
             self.template_ext = templates_exts
 
+    def __deepcopy__(self, memo):
+        """
+        Hack sorting double chained task lists by task_id to avoid hitting
+        max_depth on deepcopy operations.
+        """
+        sys.setrecursionlimit(5000)  # TODO fix this in a better way
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+
+        # since we won't mutate the arguments, we should just do the shallow copy
+        # there are some cases we can't deepcopy the objects(e.g protobuf).
+        shallow_copy = {'python_callable', 'op_kwargs'}
+        skip_deepcopy = {'user_defined_macros',
+                         'user_defined_filters',
+                         'params',
+                         '_log'} | shallow_copy
+
+        for k, v in list(self.__dict__.items()):
+            if k not in skip_deepcopy:
+                setattr(result, k, copy.deepcopy(v, memo))
+            if k in shallow_copy:
+                setattr(result, k, copy.copy(v))
+        result.params = self.params
+        if hasattr(self, 'user_defined_macros'):
```

Comment:
I think this can be done in a way where everything happens in the loop above (all parameter driven) with no per-attribute assignment.

Reply:
thanks for the info.